### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by white238**
* Default to C+11 for tests
* Fix warning that would have never triggered

### CODE REVIEW

**Comment by white238** [comment]
Wish I would have caught this before I did a release? is it worth redoing/bumping to 0.3.1?

**Comment by gzagaris** [comment]
> Wish I would have caught this before I did a release? is it worth redoing/bumping to 0.3.1?

One typical way to deal with this is to do a `hotfix` branch into `master`, increment the patch version number and tag, then merge master back to develop.

File: thirdparty_builtin/CMakeLists.txt [hunk 1]
```diff
@@ -5,6 +5,14 @@
 
 set(_blt_tpl_targets) # tracks names of enabled tpl targets
 
+if(ENABLE_GBENCHMARK OR ENABLE_GTEST OR ENABLE_GMOCK)
+  if("${CMAKE_CXX_STANDARD}" STREQUAL "" OR
+     "${CMAKE_CXX_STANDARD}" LESS 11)
+      message(WARNING "GoogleTest, GoogleMock, and GoogleBenchmark require greater than C++11. "
+                      "Set BLT_CXX_STD to 'c++11' or higher before loading BLT.")
```
Comment (gzagaris) [inline thirdparty_builtin/CMakeLists.txt]: Should this disable GTest, etc. in this case?

**Comment by rhornung67** [comment]
@white238 I think following @gzagaris suggestion is best. But, you would probably want to create a new tagged version number to avoid user confusion since some projects are already using 0.3.0.

File: thirdparty_builtin/CMakeLists.txt (hunk 1 above)
Comment (white238) [reply thirdparty_builtin/CMakeLists.txt]: We had a few discussions on this.  The concern is you don't set it but the compiler naturally supports the standard you would want it to succeed. So throw a warning but don't fail.

**Comment by white238** [comment]
So the vote is to do another release at 0.3.1 just to confirm?  I want to be clear this doesn't "fix" anything it just puts a warning that might be useful.

File: thirdparty_builtin/CMakeLists.txt (hunk 1 above)
Comment (gzagaris) [reply thirdparty_builtin/CMakeLists.txt]: Sounds good. Thanks for the clarification.

**Comment by cyrush** [comment]
if you wanted a tagged release, best to bump.

File: thirdparty_builtin/CMakeLists.txt [hunk 2]
```diff
@@ -5,6 +5,14 @@
 
 set(_blt_tpl_targets) # tracks names of enabled tpl targets
 
+if(ENABLE_GBENCHMARK OR ENABLE_GTEST OR ENABLE_GMOCK)
+  if("${CMAKE_CXX_STANDARD}" STREQUAL "" OR
+     "${CMAKE_CXX_STANDARD}" LESS 11)
+      message(WARNING "GoogleTest, GoogleMock, and GoogleBenchmark require greater than C++11. "
```
Comment (kennyweiss) [inline thirdparty_builtin/CMakeLists.txt]: Slight suggestion:
``"... require greater than C++11."`` --> ``"... require C++11 or later."``

**Pull request merged**